### DISCUSSION TITLE
Codegen subscriptions

### DISCUSSION THREAD

**Discussion by psylsph** [mirumee/ariadne#1119] [General]
Good morning to you have any idea when Subscriptions will be supported via the codegen project?

**Comment by bombsimon** [comment]
Do you mean `ariadne-codegen` and the client project? It should already be supported, from the README in `ariadne-codegen`:

> Python code generator that takes graphql schema, queries, mutations and subscriptions and generates Python package with fully typed and asynchronous GraphQL client.

> To support subscriptions, default base client requires websockets package:
> 
> ```
> $ pip install ariadne-codegen[subscriptions]
> ```

And under [websockets](https://github.com/mirumee/ariadne-codegen/tree/main#websockets):

> To handle subscriptions, default AsyncBaseClient uses [websockets](https://github.com/python-websockets/websockets) and implements [graphql-transport-ws](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) subprotocol.

There are several tests in the project that generates and uses subscriptions.

**Reply by psylsph** [reply]
My generated code is not working and one of the pages seemed to be saying it wasn’t working for codegen although the code was generating, I’m sort of convinced it’s my spring boot server that is the issue as it I manually change the sub protocol in the generated client to be graphql-ws rather than the default graphql-transport-ws I get a bit further but the server returns a ka keep alive message that the generated client rejects

**Reply by mat-sop** [reply]
@psylsph If I understand correctly, you changed the protocol name in generated base client to `graphql-ws` and received a keep-alive message but it got rejected by client?

If that's the case it looks like your server implements [deprecated `graphql-ws` protocol](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) and our base client implements [`graphql-transport-ws`](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md). These protocols are not compatible, so to make it work you need to write your own base client with support for `graphql-ws` and then use codegen with [`base_client_name` and `base_client_file_path`](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md).

**Reply by psylsph** [reply]
Thanks that's what I'm think either that or try and fix my server, the problem with the server end is it's all spring boot and auto generated and it should be talking graphql-transport-ws but refuses for some reason

**Comment by psylsph** [comment]
I think I’m going to roll my own custom client it look easy enough to integrate based on the nice configuration system
